pkg/api/errors/state.go

```diff
@@ -0,0 +1,108 @@
+	return kitErrors.New(
+		grpcCodes.FailedPrecondition,
+		http.StatusInternalServerError,
+		"state store is not configured",
+		"ERR_STATE_STORE_NOT_CONFIGURED",
+	).
+		WithErrorInfo(kitErrors.CodePrefixStateStore+kitErrors.CodeNotConfigured, nil)
+}
+
+func StateStoreNotFound(storeName string) *kitErrors.Error {
+	return kitErrors.New(
+		grpcCodes.InvalidArgument, // TODO check if it was used in the past, we should change it. It should be grpcCodes.NotFound
+		http.StatusBadRequest,
+		fmt.Sprintf("state store %s is not found", storeName),
+		"ERR_STATE_STORE_NOT_FOUND",
+	).
+		WithErrorInfo(kitErrors.CodePrefixStateStore+kitErrors.CodeNotFound, nil)
+}
+
+func StateStoreInvalidKeyName(storeName string, key string, msg string) *kitErrors.Error {
+	return kitErrors.New(
+		grpcCodes.InvalidArgument,
+		http.StatusBadRequest,
+		msg,
+		"ERR_MALFORMED_REQUEST",
+	).WithErrorInfo(kitErrors.CodePrefixStateStore+kitErrors.CodeIllegalKey, nil).
+		WithResourceInfo("state", storeName, "", "").
+		WithFieldViolation(key, msg)
+}
+
+/**** Transactions ****/
+
+func StateStoreTransactionsNotSupported(storeName string) *kitErrors.Error {
+	return kitErrors.New(
+		grpcCodes.Unimplemented,
+		http.StatusInternalServerError,
+		fmt.Sprintf(kitErrors.MsgStateTransactionsNotSupported, storeName),
+		"ERR_STATE_STORE_NOT_SUPPORTED", // TODO: @elena-kolevska this is misleading and also used for different things ("query unsupported"); it should be removed in the next major version
+	).
+		WithErrorInfo(kitErrors.CodePrefixStateStore+"TRANSACTIONS_NOT_SUPPORTED", nil).
+		WithResourceInfo("state", storeName, "", "").
+		WithHelpLink("https://docs.dapr.io/reference/components-reference/supported-state-stores/", "Check the list of state stores and the features they support")
+}
+
+func StateStoreTooManyTransactionalOps(storeName string, count int, max int) *kitErrors.Error {
+	return kitErrors.New(
+		grpcCodes.InvalidArgument,
+		http.StatusBadRequest,
+		fmt.Sprintf("the transaction contains %d operations, which is more than what the state store supports: %d", count, max),
+		"ERR_STATE_STORE_TOO_MANY_TRANSACTIONS",
+	).
+		WithErrorInfo(kitErrors.CodePrefixStateStore+"TOO_MANY_TRANSACTIONS", map[string]string{
+			"currentOpsTransaction": strconv.Itoa(count),
+			"maxOpsPerTransaction":  strconv.Itoa(max),
+		}).
+		WithResourceInfo("state", storeName, "", "")
+}
+
+/**** Query API ****/
+
+func StateStoreQueryUnsupported(storeName string) *kitErrors.Error {
+	return kitErrors.New(
+		grpcCodes.Internal,
+		http.StatusInternalServerError,
+		"state store does not support querying",
+		"ERR_STATE_STORE_NOT_SUPPORTED",
+	).
+		WithErrorInfo(kitErrors.CodePrefixStateStore+"QUERYING_"+kitErrors.CodeNotSupported, nil).
+		WithResourceInfo("state", storeName, "", "")
+}
+
+func StateStoreQueryFailed(storeName string, detail string) *kitErrors.Error {
+	// TODO: @elena-kolevska #change-in-next-major: The http and grpc codes are wrong, but they're legacy.
```

Comment:
What is wrong about the codes? We also probably will never have a new `major` version of Dapr so it will likely be this way forever :smile: